From 72ff623f9cbf2e0641d6ad19fc1f0f062e4787bb Mon Sep 17 00:00:00 2001
From: Garrett Gu <garrett@cloudflare.com>
Date: Mon, 22 Apr 2024 10:10:21 -0500
Subject: [PATCH 10/22] Modify where to look for fp16 dependency. This
 dependency is normally downloaded by gn but we need to fetch it in Bazel. We
 do so in the workerd repo in the WORKSPACE file.

---
 BUILD.bazel | 26 ++++++++++++++++----------
 1 file changed, 16 insertions(+), 10 deletions(-)

diff --git a/BUILD.bazel b/BUILD.bazel
index cf3fc8f0645..28d8993c840 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -3923,16 +3923,22 @@ v8_library(
   ],
 )
 
-v8_library(
-  name = "lib_fp16",
-  srcs = ["third_party/fp16/src/include/fp16.h"],
-  hdrs = [
-    "third_party/fp16/src/include/fp16/fp16.h",
-    "third_party/fp16/src/include/fp16/bitcasts.h",
-  ],
-  includes = [
-    "third_party/fp16/src/include",
-  ],
+cc_library(
+    name = "lib_fp16",
+    hdrs = [
+        "@fp16//:include/fp16.h",
+    ],
+    include_prefix = "third_party/fp16/src",
+    deps = [":lib_fp16_includes"],
+)
+
+cc_library(
+    name = "lib_fp16_includes",
+    hdrs = [
+        "@fp16//:include/fp16/bitcasts.h",
+        "@fp16//:include/fp16/fp16.h",
+    ],
+    strip_include_prefix = "include",
 )
 
 filegroup(
-- 
2.43.0

